pkg/api/api.go

```diff
@@ -1736,82 +1759,75 @@ func (s *ScalewayAPI) GetDashboard() (*ScalewayDashboard, error) {
 }
 
 // GetServerID returns exactly one server matching or dies
-func (s *ScalewayAPI) GetServerID(needle string) string {
+func (s *ScalewayAPI) GetServerID(needle string) (string, error) {
 	// Parses optional type prefix, i.e: "server:name" -> "name"
 	_, needle = parseNeedle(needle)
 
 	servers, err := s.ResolveServer(needle)
 	if err != nil {
-		log.Fatalf("Unable to resolve server %s: %s", needle, err)
+		return "", fmt.Errorf("Unable to resolve server %s: %s", needle, err)
 	}
 	if len(servers) == 1 {
-		return servers[0].Identifier
+		return servers[0].Identifier, nil
 	}
 	if len(servers) == 0 {
-		log.Fatalf("No such server: %s", needle)
+		return "", fmt.Errorf("No such server: %s", needle)
 	}
-
-	showResolverResults(needle, servers)
-	os.Exit(1)
-	return ""
+	return "", showResolverResults(needle, servers)
 }
 
 func showResolverResults(needle string, results ScalewayResolverResults) error {
-	log.Errorf("Too many candidates for %s (%d)", needle, len(results))
-
+	arch := ""
 	w := tabwriter.NewWriter(os.Stderr, 20, 1, 3, ' ', 0)
 	defer w.Flush()
 	sort.Sort(results)
 	for _, result := range results {
-		fmt.Fprintf(w, "- %s\t%s\t%s\n", result.TruncIdentifier(), result.CodeName(), result.Name)
+		arch = result.Arch
+		if arch == "" {
+			arch = "n/a"
+		}
+		fmt.Fprintf(w, "- %s\t%s\t%s\t%s\n", result.TruncIdentifier(), result.CodeName(), result.Name, arch)
 	}
-	return nil
+	return fmt.Errorf("Too many candidates for %s (%d)", needle, len(results))
```

Comment:
Do you have an example of the new output with the title *under* the list of candidates ?